countries/app.rb

```diff
@@ -0,0 +1,53 @@
+require 'sinatra'
+require 'sinatra/reloader'
+require 'sinatra/activerecord'
+
+set :database, { adapter: 'sqlite3', database: 'test.db' }
+
+class Question < ActiveRecord::Base
+    
+end
+
+get '/questions' do
+    @questions = Question.all
+    erb :questions
+end
+
+get '/questions/new' do
+    erb :new
+end
+
+post '/questions' do
+    question = Question.new
+    question.question = params[:question]
+    question.answer = params[:answer]
+    question.flag = params[:flag]
+    question.save
+    redirect '/questions'
+end
+
+get '/questions/edit' do
+    @question = Question.find(params [:id])
+    erb :edit
+end
+
+post '/questions/edit' do
+    question = Question.find(params [:id])
+    question.question = params [:question]
+    question.answer = params[:answer]
+    question.flag = params[:flag]
+    question.save
+    redirect '/questions'
+end
+
+get '/guess' do
+    @questions = Question.all
+    index = rand (@questons.length)
+    @question = questions[index]
+    erb :guess
+end
+
+get '/answer' do
+    @questions = Question.find (params [:id])
```

Comment:
It looks like the corresponding view is looking for the variable called `@question` (rather than `@questions`)! So computers are pretty dumb and even a small typo will really confuse them 😖 Whatever your variable is named in one place, it has to be *exactly* the same wherever else you want to use it!